backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Services/LookupService.cs

```diff
@@ -32,8 +35,22 @@ public async Task<Party> GetPartyForOrganization(string organizationNumber)
         /// <inheritdoc/>        
         public async Task<Party> GetPartyFromReporteeListIfExists(int partyId)
         {
-            Party partyInfo = await _lookupClient.GetPartyFromReporteeListIfExists(partyId);
+            Party partyInfo = await _lookupClient.GetPartyFromReporteeListIfExists(partyId);    
             return partyInfo;
         }
+
+        /// <inheritdoc/>        
+        public async Task<Party> GetPartyByUUID(Guid uuid)
+        {
+            List<Party> partyList = await _registerClient.GetPartyListByUUID(new List<Guid>() { uuid });
+            return partyList?.FirstOrDefault();
```

Comment:
I definitely should have written a comment in the code that explains this (I'll get to that), but yeah, since we only provide one party-uuid, we will only get one party back (patyUUIDs and parties map one-to-one). The reason why we use the partyList API from the registry even though we're only looking for a single party is because this list-api apparently have better performance than the one providing only the one party. So Remi told me to use this one.

The list shouldn't really be empty, I think. If no party corresponds to the partyUUID, the API should give us something else than a 200 OK status (like a NotFound or a BadRequest). However, if we somehow _do_ get an empty list, or partyList is null, the FirstOrDefault should return a default (null) and we will return null from this method, a case which is handled in the controller.

Reply:
I see, yeah then it will be fine describing it a bit more with comments :)